app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/index.js

```diff
@@ -23,13 +23,52 @@ import PipelineConfigureButton from 'components/PipelineDetails/PipelineDetailsT
 import PipelineStopButton from 'components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineStopButton';
 import PipelineRunButton from 'components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineRunButton';
 import PipelineSummaryButton from 'components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineSummaryButton';
+import { PipelineHistoryButton } from './PipelineHistoryButton';
+import { ApolloProvider } from 'react-apollo';
+import ApolloClient from 'apollo-boost';
+import { InMemoryCache, IntrospectionFragmentMatcher } from 'apollo-cache-inmemory';
+import introspectionQueryResultData from '../../../../../../graphql/fragments/fragmentTypes.json';
+import Cookies from 'universal-cookie';
+import SessionTokenStore from 'services/SessionTokenStore';
+import { useFeatureFlagDefaultFalse } from 'services/react/customHooks/useFeatureFlag';
+
+const cookie = new Cookies();
+const fragmentMatcher = new IntrospectionFragmentMatcher({
+  introspectionQueryResultData,
+});
+
+const client = new ApolloClient({
+  uri: '/graphql',
+  cache: new InMemoryCache({ fragmentMatcher }),
+  request: (operation) => {
+    if (window.CDAP_CONFIG.securityEnabled && cookie.get('CDAP_Auth_Token')) {
```

Comment:
nit: the whole block could be simplified to this
```
    let tempHeaders = {
      'Session-Token': SessionTokenStore.getState(),
      'X-Requested-With': 'XMLHttpRequest',
    }
    if (window.CDAP_CONFIG.securityEnabled && cookie.get('CDAP_Auth_Token')) {
      tempHeaders = {
        ...tempHeaders,
        authorization: `Bearer ${cookie.get('CDAP_Auth_Token')}`
      }
    }
    operation.setContext({ headers : {...tempHeaders} });
```

Reply:
Is there any way to use, or share code with, the existing `ApolloClient`? This type of initialization logic should only exist once

Reply:
I think Sean and I talked about it. The pipelineDetails view's root is angularjs, so not very straightforward to do that.

Reply:
Could we create a factory function to initialize `ApolloClient`?

Reply:
you would still be rewriting the code. The only way to share code in between the angular and react portions of the application are to put them on the window and then you would have to call window.CaskCommon.etc from the factory initializer... I'd prefer repeated code over more things on the window.